mat/symmetric.go

```diff
@@ -176,6 +197,11 @@ func (s *SymDense) reuseAs(n int) {
 	}
 }
 
+func (s *SymDense) reuseAsZeroed(n int) {
+	s.reuseAs(n)
```

Comment:
This needs to change to reuseAsNonZeroed

Reply:
I'd rather we don't do it that way but just spell out this completely.

Reply:
Actually, `useZeroed` is incorrect for the general case. So leave this as is and we'll fix `useZeroed` later. The issue is that the `zero` function is unaware of striding and so will clobber other data outside the matrix if they share a backing slice.

Reply:
Updating this, it's not fixable in itself, so this should be updated. We need some additional documentation as a warning on the `ReuseAs` methods. They are a fairly substantial footgun that cannot have a safety.

Reply:
Sorry, that was a note to myself that `SymDense.reuseAs` needs to also be changed to `SymDense.reuseAsNonZeroed`